packages/cli/test/integration/environments/source-control-import.service.test.ts

```diff
@@ -153,4 +154,141 @@ describe('SourceControlImportService', () => {
+
+	describe('if owner specified by `ownedBy` does not exist at target instance', () => {
+		it('should assign credential ownership to importing user if personal owner is not found', async () => {
+			const importingUser = await getGlobalOwner();
+
+			fsp.readFile = jest.fn().mockResolvedValue(Buffer.from('some-content'));
+
+			const CREDENTIAL_ID = nanoid();
+
+			const stub: ExportableCredential = {
+				id: CREDENTIAL_ID,
+				name: 'My Credential',
+				type: 'someCredentialType',
+				data: {},
+				ownedBy: {
+					type: 'personal',
+					personalEmail: 'test@example.com',
+				}, // user at source instance owns credential
+			};
+
+			jest.spyOn(utils, 'jsonParse').mockReturnValue(stub);
+
+			cipher.encrypt.mockReturnValue('some-encrypted-data');
+
+			await service.importCredentialsFromWorkFolder(
+				[mock<SourceControlledFile>({ id: CREDENTIAL_ID })],
+				importingUser.id,
+			);
+
+			const personalProject = await getPersonalProject(importingUser);
+
+			const sharing = await Container.get(SharedCredentialsRepository).findOneBy({
+				credentialsId: CREDENTIAL_ID,
+				projectId: personalProject.id,
+				role: 'credential:owner',
+			});
+
+			expect(sharing).toBeTruthy(); // original user missing, so importing user owns credential
+		});
+
+		it('should create a new team project if credential owning project is not found', async () => {
+			const importingUser = await getGlobalOwner();
+
+			fsp.readFile = jest.fn().mockResolvedValue(Buffer.from('some-content'));
+
+			const CREDENTIAL_ID = nanoid();
+
+			const stub: ExportableCredential = {
+				id: CREDENTIAL_ID,
+				name: 'My Credential',
+				type: 'someCredentialType',
+				data: {},
+				ownedBy: {
+					type: 'team',
+					teamId: '1234-asdf',
+					teamName: 'Marketing',
+				}, // user at source instance owns credential
+			};
+
+			jest.spyOn(utils, 'jsonParse').mockReturnValue(stub);
+
+			cipher.encrypt.mockReturnValue('some-encrypted-data');
+
+			{
+				const project = await Container.get(ProjectRepository).findOne({
+					where: [
+						{
+							id: '1234-asdf',
+						},
+						{ name: 'Marketing' },
+					],
+				});
+
+				expect(project?.id).not.toBe('1234-asdf');
+				expect(project?.name).not.toBe('Marketing');
+			}
+
+			await service.importCredentialsFromWorkFolder(
+				[mock<SourceControlledFile>({ id: CREDENTIAL_ID })],
+				importingUser.id,
+			);
+
+			const sharing = await Container.get(SharedCredentialsRepository).findOne({
+				where: {
+					credentialsId: CREDENTIAL_ID,
+					role: 'credential:owner',
+				},
+				relations: { project: true },
+			});
+
+			expect(sharing?.project.id).toBe('1234-asdf');
+			expect(sharing?.project.name).toBe('Marketing');
+
+			expect(sharing).toBeTruthy(); // original user missing, so importing user owns credential
+		});
+	});
+
+	describe('if owner specified by `ownedBy` does exist at target instance', () => {
+		it('should use the existing team project if credential owning project is found', async () => {
```

Comment:
Maybe we can add a test that makes sure this won't re-own the credential in case it exists in the source and target with different owners. Something like this:

```ts
		it('should not change the owner if the credential is owned by somebody else on the target instance', async () => {
			cipher.encrypt.mockReturnValue('some-encrypted-data');

			const importingUser = await getGlobalOwner();

			fsp.readFile = jest.fn().mockResolvedValue(Buffer.from('some-content'));

			const targetProject = await createTeamProject('Marketing');
			const credential = await saveCredential(randomCredentialPayload(), {
				project: targetProject,
				role: 'credential:owner',
			});

			const sourceProjectId = nanoid();

			const stub: ExportableCredential = {
				id: credential.id,
				name: 'My Credential',
				type: 'someCredentialType',
				data: {},
				ownedBy: {
					type: 'team',
					teamId: sourceProjectId,
					teamName: 'Sales',
				},
			};

			jest.spyOn(utils, 'jsonParse').mockReturnValue(stub);

			await service.importCredentialsFromWorkFolder(
				[mock<SourceControlledFile>({ id: credential.id })],
				importingUser.id,
			);

			await expect(
				Container.get(SharedCredentialsRepository).findBy({
					credentialsId: credential.id,
				}),
			).resolves.toMatchObject([
				{
					projectId: targetProject.id,
					role: 'credential:owner',
				},
			]);
			await expect(
				Container.get(CredentialsRepository).findBy({
					id: credential.id,
				}),
			).resolves.toMatchObject([
				{
					name: stub.name,
					type: stub.type,
					data: 'some-encrypted-data',
				},
			]);
		});
```